src/utils.c

```diff
@@ -356,3 +356,20 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// Adapted from the bit64 package C function as_character_integer64
+// Allows us to be integer64 aware without the user loading the bit64 package
+SEXP asCharacterInteger64(SEXP x) {
+  SEXP coerced = PROTECT(allocVector(STRSXP, length(x)));
```

Comment:
I would write `xlength` instead of `length` . I would also put it in a `const int64_t` variable given that using it twice. I would also write `int64_t i = 0` in the `for` loop below.